rasa/utils/tensorflow/temp_keras_modules.py

```diff
@@ -179,28 +372,40 @@ def fit(
                 callbacks.on_epoch_begin(epoch)
                 with data_handler.catch_stop_iteration():
                     for step in data_handler.steps():
-                        with trace.Trace(
-                            "TraceContext",
-                            graph_type="train",
+                        with tf.profiler.experimental.Trace(
+                            "train",
                             epoch_num=epoch,
                             step_num=step,
                             batch_size=batch_size,
+                            _r=1,
                         ):
                             callbacks.on_train_batch_begin(step)
-                            tmp_logs = train_function(iterator)
+                            tmp_logs = self.train_function(iterator)
                             if data_handler.should_sync:
                                 context.async_wait()
                             logs = tmp_logs  # No error, now safe to assign to logs.
                             end_step = step + data_handler.step_increment
                             callbacks.on_train_batch_end(end_step, logs)
+                            if self.stop_training:
+                                break
+
                 logs = tf_utils.sync_to_numpy_or_python_type(logs)
+                if logs is None:
+                    raise ValueError(
+                        "Unexpected result of `train_function` "
+                        "(Empty logs). Please use "
+                        "`Model.compile(..., run_eagerly=True)`, or "
+                        "`tf.config.run_functions_eagerly(True)` for more "
+                        "information of where went wrong, or file a "
+                        "issue/bug to `tf.keras`."
+                    )
                 epoch_logs = copy.copy(logs)
 
                 # Run validation.
                 if validation_data and self._should_eval(epoch, validation_freq):
                     # Create data_handler for evaluation and cache it.
                     if getattr(self, "_eval_data_handler", None) is None:
-                        self._eval_data_handler = CustomDataHandler(
+                        self._eval_data_handler = data_adapter.get_data_handler(
```

Comment:
This was kind of a copy and paste bug on my side but -- now that I see it I think we should leave it as is. If we ever make use of this, we wouldn't use the custom data handler with the increasing batch size, no?

Reply:
Yeah, this seems correct